pkg/genericapiserver/genericapiserver.go

```diff
@@ -475,6 +479,10 @@ func (s *GenericAPIServer) init(c *Config) {
 
 	attributeGetter := apiserver.NewRequestAttributeGetter(s.RequestContextMapper, s.NewRequestInfoResolver())
 	handler = apiserver.WithAuthorizationCheck(handler, attributeGetter, s.authorizer)
+	if c.EnableAudit {
+		// audit handler must comes before the impersonationFilter to read the original user
+		handler = audit.WithAudit(handler, s.RequestContextMapper)
```

Comment:
Should this come before authorization? I imagine we'd want to capture authentication requests that were denied for some reason or another.